deeppavlov/models/ranking/siamese_model.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from typing import List, Iterable, Union
+
+from deeppavlov.core.models.nn_model import NNModel
+
+
+class SiameseModel(NNModel):
+    """The class implementing base functionality for siamese neural networks.
+
+    Args:
+        batch_size: A size of a batch.
```

Comment:
Why do you need a batch size that is different from the one that is set in config's `train` block? This should be described in the docstring and, maybe, in the parameter name

Reply:
I need the same batch size as in the `train` block but I can't pass it to the model.